de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/dvlm/mat/MatExporter.java

```diff
@@ -0,0 +1,288 @@
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ReferencePropertyInstance
+	 * only uuid and fullQualifiedInstanceName are saved
+	 * @param element ReferencePropertyInstance
+	 */
+	private Array contentOfProperty(ReferencePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof ReferenceProperty) {
+			ATypeInstance referencedTypeInstance = element.getReference();
+			if (element.getReference() != null) {
+				struct.set(MatHelper.UUID, Mat5.newString(referencedTypeInstance.getUuid().toString()));
+				struct.set(MatHelper.FULLNAME, Mat5.newString(referencedTypeInstance.getFullQualifiedInstanceName()));
+			} else {
+				struct.set(MatHelper.UUID, Mat5.newString(""));
+				struct.set(MatHelper.FULLNAME, Mat5.newString(""));
+			}
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ResourcePropertyInstance
+	 * only uri is saved
+	 * @param element ResourcePropertyInstance
+	 */
+	private Array contentOfProperty(ResourcePropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof ResourceProperty) {
+			BeanPropertyResource bpr = new BeanPropertyResource(element);
+			struct.set(MatHelper.URI, (!bpr.isSet()) ? Mat5.newString("") : Mat5.newString(bpr.getValue().toString()));
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about EnumUnitPropertyInstance
+	 * only unit,value and name are saved
+	 * @param element EnumUnitPropertyInstance
+	 */
+	private Array contentOfProperty(EnumUnitPropertyInstance element) {
+		Struct struct = Mat5.newStruct();
+		if (element.getType() instanceof EnumProperty) {
+			BeanPropertyEnum bpe = new BeanPropertyEnum(element);
+			struct.set(MatHelper.UNIT, Mat5.newString(bpe.getUnit()));
+			struct.set(MatHelper.VALUE, (!bpe.isSet()) ? Mat5.newString("") : Mat5.newScalar(bpe.getEnumValue()));
+			struct.set(MatHelper.NAME, (!bpe.isSet()) ? Mat5.newString("") : Mat5.newString(bpe.getValue()));
+		}
+		return struct;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ArrayInstance
+	 * @param element ArrayInstance
+	 */
+	private Array contentOfProperty(ArrayInstance element) {
+		EList<APropertyInstance> propertyInstances = element.getArrayInstances();
+		Cell cell = Mat5.newCell(propertyInstances.size(), 1);
+		for (int i = 0; i < propertyInstances.size(); i++) {
+			APropertyInstance instanceElement = propertyInstances.get(i);
+			Array propertyArray = getRightProperty(instanceElement);
+			cell.set(i, propertyArray);
+		}
+		return cell;
+	}
+
+	/**
+	 * creates a new .mat-struct with all information about ComposedPropertyInstance
+	 * @param element ComposedPropertyInstance
+	 */
+	private Array contentOfProperty(ComposedPropertyInstance element) {
+		return exportCatAs(element.getTypeInstance());
+	}
+
+	//Delete First and Last Character of a string.

```

Comment:
Method comments are usually done with `/** **/`
Also could use a @param str in the comment.